docs/docs/jsonnet/native-functions.md

```diff
@@ -6,65 +6,216 @@ menu: "Writing Jsonnet"
 
 # Native Functions
 
-Tanka extends the Jsonnet standard library with some common helper utilities.
+Native functions are Jsonnet functions implemented in Go rather than Jsonnet itself. Tanka extends the Jsonnet native functions with some common helper utilities.
 
-## parseJSON
+To use a native function in your Jsonnet code, you need to use the `native` function from the Jsonnet standard library. For example, to parse a Kubernetes deployment YAML file into a Jsonnet object:
 
-```ts
-parseJSON(string json) Object
+```jsonnet
+{
+  deployment:  std.native('parseYaml')(importstr 'deployment.yaml'),
+}
 ```
 
-`parseJSON` wraps `json.Unmarshal` to convert a json string into a dict
+The argument to `std.native` is the name of the native function to lookup. In this case, it is `parseYaml`. The returned function is then evaluated with the arguments in the second set of parentheses. This is the `deployment.yaml` file imported as a string using the Jsonnet keyword `importstr`.
 
-## parseYAML
+## parseJson
 
-```ts
-parseYAML(string yaml) []Object
+### Usage
+
+```jsonnet
+std.native('parseJson')(json)
```

Comment:
I'd prefer to revert back to the function signatures, as they can come handy as reference material, especially regarding argument types.

When replacing the `Usage` section with `Signature`, your `Examples` section should still provide enough information on how to use it!

Reply:
That's true, I did hesitate when replacing the signature with the usage because losing the typing did make the native functions arguments less clear and the workaround of mentioning the types in the subsequent comment felt a bit clunky. Like in the `regexSubst` comment:
```
regexSubst` replaces all matches of the re2 regular expression `regex` in the `src` string with the `repl` string.
```

I'll make the switch back to `Signature` and revert the associated comments too.